affordance_util/src/affordance_util_tester.cpp

```diff
@@ -95,20 +95,72 @@ int main()
     std::cout << "Input T: \n" << T << std::endl;
     std::cout << "Output: \n" << MatrixLog6(T) << std::endl;
 
-    /* const std::string filepath = "/home/crasun/spot_ws/src/cc_affordance_planner/config/robot_setup.yaml"; */
-    /* const RobotConfig robot_config = robot_builder(filepath); */
-    /* std::cout << "\n Testing robot_builder" << std::endl; */
-    /* std::cout << "Input filepath: \n" << filepath << std::endl; */
-    /* std::cout << "Output Slist: \n" << robot_config.Slist << std::endl; */
-    /* std::cout << "Output M: \n" << robot_config.M << std::endl; */
-    /* std::cout << "Output ref_frame_name: \n" << robot_config.ref_frame_name << std::endl; */
-    /* std::cout << "Output tool_name: \n" << robot_config.tool_name << std::endl; */
-    /* std::cout << "Output joint_names: "; */
-    /* for (const std::string &joint_name : robot_config.joint_names) */
-    /* { */
-    /*     std::cout << joint_name << ","; */
-    /* } */
-    /* std::cout << std::endl; */
+    std::cout << "\nTesting robot_builder YAML version" << std::endl;
+    std::string filepath = "/home/john/random_tests/temp_cca_ws/cca_kinova_gen3_7dof_description.yaml";
+    RobotConfig robot_config;
+    try
+    {
+        robot_config = robot_builder(filepath);
+    }
+    catch (const std::exception &e)
+    {
+        std::cerr << "Error building robot: " << e.what() << std::endl;
+    }
+    std::cout << "Input filepath: \n" << filepath << std::endl;
+    const Eigen::MatrixXd yaml_slist = robot_config.Slist;
+    std::cout << "Output Slist: \n" << yaml_slist << std::endl;
+    std::cout << "Output M: \n" << robot_config.M << std::endl;
+    std::cout << "Output ref_frame_name: \n" << robot_config.ref_frame_name << std::endl;
+    std::cout << "Output tool_name: \n" << robot_config.tool_name << std::endl;
+    std::cout << "Output joint_names: ";
+    for (const std::string &joint_name : robot_config.joint_names)
+    {
+        std::cout << joint_name << ",";
+    }
+    std::cout << std::endl;
+    filepath = "/home/john/temp_cca_ws/GEN3_URDF_V12.urdf";
+    const std::string ref_frame = "base_link";
+    const std::string base_joint = "Actuator1";
+    const std::string ee_frame = "EndEffector_Link";
+    // const std::string filepath = "/home/john/temp_cca_ws/spot_arm.urdf";
+    // const std::string ref_frame = "spot_arm_base_link";
+    // const std::string base_joint = "spot_arm_shoulder_yaw";
+    // const std::string ee_frame = "spot_arm_fingers";
+    try
+    {
+        robot_config = robot_builder(filepath, ref_frame, base_joint, ee_frame);
```

Comment:
Also test the version with the tool_location argument. Do this for both Kinova and Spot and compare with the yaml versions using a similar approach as slist. They should match within tolerance.